kubernetes-azure-go/main.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"encoding/base64"
+
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/containerservice"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/network"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/resources"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		// Get some configuration values or set default values
+		cfg := config.New(ctx, "")
+		azureLocation := config.Require(ctx, "azure-native:location")
+		sshPubKey := cfg.Require("sshPubKey")
+		mgmtGroup := cfg.Require("mgmtGroupId")
+		prefixForDns, err := cfg.Try("prefixForDns")
+		if err != nil {
+			prefixForDns = "pulumi"
+		}
+		kubernetesVersion, err := cfg.Try("kubernetesVersion")
+		if err != nil {
+			kubernetesVersion = "1.24.3"
+		}
+		numWorkerNodes, err := cfg.TryInt("numWorkerNodes")
+		if err != nil {
+			numWorkerNodes = 3
+		}
+		nodeVmSize, err := cfg.Try("nodeVmSize")
+		if err != nil {
+			nodeVmSize = "Standard_DS2_v2"
+		}
+
+		// Create an Azure Resource Group
+		resourceGroup, err := resources.NewResourceGroup(ctx, "resourceGroup", &resources.ResourceGroupArgs{
+			Location:          pulumi.String(azureLocation),
+			ResourceGroupName: pulumi.String("aks-rg"),
+		})
+		if err != nil {
+			return err
+		}
+		ctx.Export("resourceGroupName", resourceGroup.Name)
```

Comment:
This is mostly anecdotal, but I think I've generally seen these clustered up at the end of the program, rather than sprinkled throughout, and this is generally how I've written my programs as well. Do you think we should be consistent there, or do you prefer this approach to that one?

Reply:
Fixed via 270555d